Cannot generate Ninja files with CMake

<!--
Use this issue if you found a bug in the preCICE library. You can also open issues in [preCICE adapters and tutorials](https://github.com/precice).
For general user support, please use the [Discourse forum](https://precice.discourse.group/) or the [Gitter chatroom](https://gitter.im/precice/Lobby).
-->

**Describe your setup**

Operating system (e.g. Linux distribution and version): _Arch Linux_
preCICE Version: 2.5.0

Last commit:
a7ae545c9b8a8b228180031eb6eb93c24f610c61

**Describe the problem**
Cannot generate Ninja files with CMake. 
Unix Makefiles are working fine.

**Step To Reproduce**
``` sh
mkdir build
cd build
cmake -GNinja  ..
```
Gives this error message:
```
CMake Error:
  Running

   '/usr/bin/ninja' '-C' '/home/tan/master/advanced_programming/project/precice/build' '-t' 'recompact'

  failed with:

   ninja: warning: phony target 'binprecice' names itself as an input; ignoring [-w phonycycle=warn]

  ninja: error: build.ninja:4102: multiple rules generate binprecice
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -369,7 +369,6 @@ if (PRECICE_BUILD_TOOLS)
     COMMAND ${CMAKE_COMMAND} -E create_symlink precice-tools binprecice
     DEPENDS precice-tools
     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-    BYPRODUCTS binprecice
     )
 endif()
 

